src/scripts/main.js

```diff
@@ -1,5 +1,45 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const parentElement = spider.parentElement;
+  const parentRect = parentElement.getBoundingClientRect();
+
+  const borderWidth = parentRect.width - parentElement.clientWidth;
+  const borderHeight = parentRect.height - parentElement.clientHeight;
```

Comment:
The calculation of 'borderWidth' and 'borderHeight' might be unnecessary if the intention is to simply keep the spider within the visible area of its parent. These calculations are typically used to account for borders, but if the parent element has no borders, these values will be zero and can be omitted. Verify if the parent element has borders and adjust the logic if needed.